Add additional Romi Hardware information (#1386)
wpilibsuite/frc-docs: source/docs/romi-robot/hardware-support.rst

Romi Hardware Support
=====================

> The Romi robot has limited hardware interfacing support. It is not compatible with the large majority of FRC based control system components. Details on the pin mapping is available in :doc:`getting-to-know-romi`.

Reviewer: ```suggestion
The Romi robot, having a different hardware architecture as a roboRIO, is compatible with a subset of commonly used FRC control system components.

Compatible Hardware
------------------------

In general, the Romi is compatible with the following:

- Simple Digital Input/Output devices (e.g. bumper switches, single LEDs)
- Standard RC-style PWM output devices (e.g. servos, PWM based motor controllers)
- Analog Input sensors (e.g distance sensors that report distance as a voltage)
```

Incompatible Hardware
---------------------

> The Romi Robot DIO is not compatible with the following:

Reviewer: ```suggestion
Due to hardware limitations, the Romi Robot is not compatible with the following:
```

- Non-integrated encoders

> - Ultrasonics

Reviewer: ```suggestion
- "Ping" style ultrasonic sensors (which require 2 DIO channels)
```

> - Timing based sensors

Reviewer: ```suggestion
- Timing based sensors
- CAN based devices
```

Romi Hardware Support
=====================

The Romi robot, having a different hardware architecture as a roboRIO, is compatible with a subset of commonly used FRC control system components.

Compatible Hardware
------------------------

In general, the Romi is compatible with the following:

- Simple Digital Input/Output devices (e.g. bumper switches, single LEDs)
- Standard RC-style PWM output devices (e.g. servos, PWM based motor controllers)
- Analog Input sensors (e.g distance sensors that report distance as a voltage)

Incompatible Hardware
---------------------

Due to hardware limitations, the Romi Robot is not compatible with the following:

> - Non-integrated encoders

Reviewer: ```suggestion
- Encoders other than the Romi-integrated encoders
```
